### PR TITLE
ENH: TimedeltaArray add/sub with NaT preserve reso

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -2105,3 +2110,21 @@ def maybe_infer_freq(freq):
...
+    dtype : DatetimeTZDtype or np.dtype
+        If np.dtype, we assume it is a datetime64 dtype.
+
+    Returns
+    -------
+    str
+    """
+    if isinstance(dtype, DatetimeTZDtype):
+        return dtype.unit
+    return str(dtype).split("[")[-1][:-1]
```
Comment (mroeschke) [inline pandas/core/arrays/datetimelike.py:2130]: `datetime_data` is probably the more "official" way to get the unit from an np.dtype: https://numpy.org/devdocs/reference/generated/numpy.datetime_data.html

File: pandas/tests/arrays/test_timedeltas.py [hunk 2]
```diff
@@ -33,39 +41,55 @@ def test_non_nano(self, unit, reso):
...
 
         result = tda.total_seconds()
         expected = tda_nano.total_seconds()
         tm.assert_numpy_array_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "nat", [np.datetime64("NaT", "ns"), np.datetime64("NaT", "us")]
+    )
+    def test_add_nat_datetimelike_scalar(self, nat, tda):
+        result = tda + nat
```
Comment (mroeschke) [inline pandas/tests/arrays/test_timedeltas.py:72]: Is the sub behavior already tested?

Comment (jbrockmendel) [reply pandas/tests/arrays/test_timedeltas.py:72]: no, separate PR

**Comment by jbrockmendel** [comment]
updated + green

**Review by mroeschke — APPROVED** [review]

**Pull request merged**